dotCMS/src/main/java/com/dotmarketing/business/RelationshipAPIImpl.java

```diff
@@ -451,14 +455,22 @@ public void convertRelationshipToRelationshipField(final Relationship oldRelatio
                 oldRelationship.getCardinality(), oldRelationship.isChildRequired(),
                 childContentType.variable());
 
-        createRelationshipField(parentFieldName, childContentType.id(),
+        final com.dotcms.contenttype.model.field.Field childRelationshipField =createRelationshipField(parentFieldName, childContentType.id(),
                 oldRelationship.getCardinality(), oldRelationship.isParentRequired(),
                 parentContentType.variable() + "." + parentRelationshipField.variable());
 
-        //Get the new Relationship
-        final Relationship newRelationship = byTypeValue(
-                parentContentType.variable() + "." + parentRelationshipField.variable());
 
+        Relationship newRelationship = (Relationship) Try.of(()->  BeanUtils.cloneBean(oldRelationship)).getOrElseThrow(e-> new DotRuntimeException(e));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'newRelationship' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5653044)